TestProject/ClientApp/src/components/Posts.js

```diff
@@ -3,11 +3,11 @@ import { NotificationContainer, NotificationManager } from 'react-notifications'
 import Post from './Post';
 import { getRating } from '../services/RatingService'; 
 
-function Posts({ userProfile, filterByCurrentUser = false }) {
+function Posts({ currentUser, filterByUser = false, authorId }) {
```

Comment:
It is filtering property, you don't need a flag for every of them. Now there is only one, but can be many.
Checking `authorId` on `falsy`ness is enough to return all without filtering, you don't even need to check it here, the backend can handle all the filter parameters.
You can put all filtering options in a single `filter` object when they will be more than one of them.